### DISCUSSION TITLE
Custom Theme vs  Custom  Layout

### DISCUSSION THREAD

**Discussion by earthaYan** [vuejs/vitepress#3549] [Q&A]
what is different between them?and does Custom Theme include Custom  Layout?

**Comment by brc-dd** [comment]
Custom layouts are only for default theme. Basically useful if you want to override the layout on page by page basis (you'll need to specify layout using frontmatter on each page).

Custom theme on the other hand is like completely different theme. You can do whatever you want there.

**Comment by subaru-git** [comment]
## 議題: 募集作成画面でメンバー再追加時の `isAttendable` 初期値

### 実装PR

https://github.com/wantedly/wantedly-admin-frontend/pull/6007/commits/626c95bf13353d34aaeedd70c38d8b0ce89bb1d7

### 現状の挙動

募集作成画面（ステップ5）の指名設定をチェックボックス方式に変更しました。`staffing.isAttendable` を GraphQL fragment で取得し、ローカル state で管理しています。

メンバー追加時（`addMember` / `addEmployeeInvitation`）に新規 staffing オブジェクトを作成しますが、この時 `isAttendable` の初期値は `false` に固定しています。

### 問題点

**既存募集の編集時**に、同一セッション内で以下の操作を行うと不整合が発生します：

1. メンバーAの指名設定を ON にする
2. メンバーAを staffing から削除する
3. メンバーAを再度追加する
4. → `isAttendable: false` の新規 staffing が作られるため、**ローカルでは OFF** だが **サーバー上は ON** のまま

※ 現時点では `StaffingInput` にまだ `isAttendable` が含まれていないため保存はされませんが、サーバー側拡張後に顕在化します。

なお、ページリロードやステップ移動で fragment から再取得すればサーバーの値が正しく反映されるため、**問題が起きるのは同一セッション内での削除→再追加のみ**です。

### 対応案

| 案 | 内容 | メリット | デメリット |
|---|------|---------|----------|
| A | 再追加時にサーバーから既存の `isAttendable` を取得して初期値にする | 正確 | 追加の API コールが必要 |
| B | `false` 固定のまま、ユーザーに再設定してもらう | シンプル | 体験が劣る、意図せず OFF で保存されるリスク |
| C | 削除した staffing の `isAttendable` をローカルにキャッシュしておき、再追加時に復元する | API コール不要 | 実装の複雑化 |

現状は案Bで実装しています。サーバー側の `StaffingInput` 拡張のタイミングで判断が必要になるため、共有しておきます。
